src/fetchClient.ts

```diff
@@ -0,0 +1,131 @@
+import { ResponseType } from './responseType';
+
+import { FetchClientConfiguration, RequestOptions } from './index';
+type RequestParameters = Record<string, string | number>;
+
+class FetchClient {
+  private configuration: FetchClientConfiguration = {
+    baseUrl: '',
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    headers: { 'Content-Type': 'application/json' },
+  };
+
+  private getPath({
+    path,
+    options,
+  }: {
+    path: string;
+    params?: RequestParameters;
+    options?: RequestOptions;
+  }) {
+    const baseUrl = options?.baseUrl || this.configuration.baseUrl;
+    if (!baseUrl) {
+      throw new Error(
+        'FetchClient is not configured. Please run initApiClient() or pass a custom baseUrl.'
+      );
+    }
+    return [
+      baseUrl.endsWith('/') ? baseUrl.slice(0, -1) : baseUrl,
+      path.startsWith('/') ? path.slice(1) : path,
+    ].join('/');
+  }
+
+  private getHeaders(options?: RequestOptions): Record<string, string> {
+    return {
+      ...this.configuration.headers,
+      ...options?.headers,
+      ...(options?.accessToken
+        ? { Authorization: `Bearer ${options.accessToken}` }
+        : {}),
+    };
+  }
+
+  private static async returnData(response: Response): Promise<ResponseType> {
```

Comment:
I would avoid returning custom responses instead of fetching and cascading potential errors.

Reply:
what you are suggesting is to return the whole response with a parsed body instead of only parts of it, right? I'm ok with that. What do you think @Averethel ?

Reply:
Yes. I would have to read the docs in detail but basically, there can be other sources of errors than just an API error for exampel

Reply:
I'm okay with returning the raw response type with a parsed body, provided we can type the body.
I think it's perfectly fine not to raise if the API returned an error and handle different types of errors (eg. network) via exceptions. In general, exception handling is expensive (think about stack traces and new contexts etc.) so I would avoid exceptions driven error handling whenever possible.

Reply:
As discussed, we return the complete response now with a parsed and typed body 🎉